framework/include/problems/FEProblemBase.h

```diff
@@ -971,6 +971,36 @@ class FEProblemBase : public SubProblem, public Restartable
   bool vectorPostprocessorHasVectors(const std::string & vpp_name);
   ///@}
 
+  /**
+   * Get a read-only reference to a reporterv alue.
+   * @param reporter_name The name of the reporter value (object_name/value_name).
+   * @return Reference to the data.
+   *
+   * Note: This method is only for retrieving values that already exist, the Reporter and
+   *       ReporterInterface objects should be used rather than this method for creating
+   *       and getting values within objects.
+   */
+  template <typename T>
+  const T & getReporterValueByName(const ReporterName & reporter_name,
+                                   std::size_t t_index = 0) const
+  {
+    return _reporter_data.getReporterValue<T>(reporter_name, t_index);
+  }
```

Comment:
I am not sure I am sold on having this function. This really isn't something that I want to be easy to do, applications should use the interface classes. Can you just make the Transfer call the const (non-key) version of getReporterData()?

Reply:
Unfortunately, I don't think we can use the interface class for the transfer because it needs to get data from the sub applications. From what I can tell, the interface class can only recognizes the application from which it is built, i.e. the main application. I can make another interface class for getting data from applications, but this seems really specific for this transfer.

Reply:
Correct. My comment wasn't clear. I was trying to say that I want applications to use the interfaces and this helper will make it more tempting to not use the interface.  The transfer could call problem.getReporterData().getReporterValue() on the sub-applications.

Reply:
To your second point. This is basically just a utility bypassing `getReporterData()`.  I have no problem deleting it.

Reply:
Yup, but I think having the bypass makes it too easy to bypass the interfaces. I think you only call this in two spots, so for now I would like to avoid the helper. Thanks.